camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/KafkaReader.java

```diff
@@ -116,6 +116,8 @@ public boolean getNext(EtlKey key, BytesWritable payload, BytesWritable pKey) th
         bytes = new byte[origSize];
         buf.get(bytes, buf.position(), origSize);
         pKey.set(bytes, 0, origSize);
```

Comment:
Can you add a logging when the key is null so we know this situation happened.

And when you set the pKey.setSize(0), does that cause any problems in the remaining code?